data/net/minecraft/world/level/levelgen/placement/PlacementContext.mapping

```diff
@@ -3,3 +3,12 @@ CLASS net/minecraft/world/level/levelgen/placement/PlacementContext
 		ARG 1 level
 		ARG 2 generator
 		ARG 3 topFeature
+	METHOD getBlockState (Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/block/state/BlockState;
+		ARG 1 pos
+	METHOD getCarvingMask (Lnet/minecraft/world/level/ChunkPos;Lnet/minecraft/world/level/levelgen/GenerationStep$Carving;)Lnet/minecraft/world/level/chunk/CarvingMask;
+		ARG 1 chunkPos
+		ARG 2 carving
```

Comment:
```suggestion
		ARG 2 step
```

This is either `AIR` or `LIQUID`, it's being used to get a specific carving mask (the noun) of a given (generation) step. "carving" doesn't refer to anything.